### PR TITLE
NEON pipeline infrastructure refactoring

### PR DESCRIPTION
**Pull request by sujaypatil96**
Apologies this PR took so long for me to open up for review @pkalita-lbl.

You had left a few comments on your fantastic review of  https://github.com/microbiomedata/nmdc-runtime/pull/308, and I had said that I would address those comments on a separate PR, so here's that PR.

Absolutely no rush, but could you review this when you have some time?

### CODE REVIEW

File: tests/test_data/test_neon_benthic_data_translator.py [hunk 1]
```diff
@@ -130,52 +128,36 @@
     ),
 }
 
+def neon_envo_mappings_file():
```
Comment (sujaypatil96) [inline tests/test_data/test_neon_benthic_data_translator.py:131]: Addressing your comment about making live network requests in tests: https://github.com/microbiomedata/nmdc-runtime/pull/308#discussion_r1416066358

I've added mock data to the test files directly.

File: nmdc_runtime/site/ops.py [hunk 2]
```diff
@@ -872,3 +905,41 @@ def get_csv_rows_from_url(url: Optional[str]) -> List[Dict]:
...
+    response.raise_for_status()
+
+    # Using Pandas read_csv to directly read the file-like object
+    df = pd.read_csv(url, delimiter="\t")
+
+    return df
+
+
+@op
+def site_code_mapping() -> dict:
```
Comment (sujaypatil96) [inline nmdc_runtime/site/ops.py:930]: Separate `@op` that returns a dictionary of site codes (as key) and their constructed `geo_loc_name` (as value).

In response to: https://github.com/microbiomedata/nmdc-runtime/pull/308#discussion_r1416102634

File: nmdc_runtime/site/ops.py [hunk 3]
```diff
@@ -872,3 +905,41 @@ def get_csv_rows_from_url(url: Optional[str]) -> List[Dict]:
     # Collect all the rows into a list of dicts while stripping out (valfilter) cells where the
     # value is an empty string (identity returns a Falsy value).
     return [valfilter(identity, row) for row in reader]
+
+
+@op
+def get_df_from_url(url: str) -> pd.DataFrame:
```
Comment (sujaypatil96) [inline nmdc_runtime/site/ops.py:911]: `@op` utilities used in the appropriate `@graph`. Basically abstracting out functionality that previously used to exist in the Translator, and using Dagster infrastructure to implement data source I/O and making sure the Translator purely has _translation_ logic.

In response to: https://github.com/microbiomedata/nmdc-runtime/pull/308#discussion_r1416102763

**Review by sujaypatil96 — COMMENTED** [review]
Adding a few comments referencing the previous review.

**Review by pkalita-lbl — APPROVED** [review]
Looks good!

**Pull request merged**